### ISSUE TITLE
math.min/math.max and NaN

### ISSUE THREAD

**Issue by fckxorg** [LuaJIT/LuaJIT#957]
Even after the patch https://github.com/LuaJIT/LuaJIT/commit/03208c8162af9cc01ca76ee1676ca79e5abe9b60 ("Fix math.min()/math.max() inconsistencies."), the following script is still faulty on x86_64:

```
LUA_PATH="src/?.lua;;" src/luajit -Ohotloop=1 -e '
local min = math.min
local max = math.max
local nan = 0/0

local r_assoc = {}
local r_assoc2 = {}
print("MIN - MAX:")
for i = 1, 4 do
    r_assoc[i] = min(1, max(nan, 1))
    r_assoc2[i] = min(1, max(1, nan))
end
print(r_assoc[1], r_assoc[2], r_assoc[3], r_assoc[4])
print(r_assoc2[1], r_assoc2[2], r_assoc2[3], r_assoc2[4], "\n")

print("MAX - MIN:")
for i = 1, 4 do
    r_assoc[i] = max(1, min(nan, 1))
    r_assoc2[i] = max(1, min(1, nan))
end
print(r_assoc[1], r_assoc[2], r_assoc[3], r_assoc[4])
print(r_assoc2[1], r_assoc2[2], r_assoc2[3], r_assoc2[4], "\n")
```

Output:
```
MIN - MAX:
1 1 1 1
nan nan nan 1

MAX - MIN:
1 1 1 1
nan nan 1 1
```

**Comment by MikePall** [comment]
It's not 'faulty' -- the behavior of `math.min/math.max` wrt. NaNs is undefined and architecture-dependent.

See [this mailing list posting](https://www.freelists.org/post/luajit/LuaJIT-behavior-regarding-NaN-in-minmax,1) for an explanation and a solution:
> If you need some specific behavior, implement the functionality
> yourself with comparisons (yes, they behave consistently) and take
> the performance hit.

**Comment by Buristan** [comment]
Fixed via 7a2b83a0c5d980bf3db0aeda33c79e7bb4b3da01. See #1082 for details.
